### ISSUE TITLE
Setters not allowed

### ISSUE THREAD

**Issue by simonbruneaud** [meltingice/psd.js#94]
Hi,

It's possible to enable setters on the tree structure model.
I could be great to add field and to manage the structure on certain cases.

Best

**Comment by fabulousduck** [comment]
What kind of setters did you have in mind for `managing` the fields ?
As for adding fields, i think that would be a little beyond the scope of this project as this is only a parser. 

@meltingice, what do you think of the idea of adding functionalities for directly manipulating the PSD object ?

**Comment by meltingice** [comment]
Modifying PSDs is beyond the scope of the project. Not sure what kind of setters you have in mind.

**Comment by simonbruneaud** [comment]
Yes you're true its a parser, I was thinking if extra properties for each node.

It could be nice to have extra informations about Layer types for instance. Currently you have to parse multiples properties to know the type. I also don't know if there is à way to extract smart objects.

**Comment by fabulousduck** [comment]
@simonbruneaud For the idea that we could get the "type" of a layer, we would have to always "assume" what a layer is as layers don't have set "types". the only thing we could guess with any amount of certainty would be texts.
In my opinion, only giving type to one node out of all of them does not make sense because it would make it highly inconsistent and would still require a check if the property exists on the node which would make you end up still have to check yourself.

@meltingice what do you think of giving layers "types" ?

As for smart objects, I will take a look for you to see if this does not already exists as I myself am not completely sure to what extent they are implemented.

**Comment by meltingice** [comment]
Beyond parsing the open/closed folder or layer property, there's no way to deduce a more specific layer type without parsing a bunch of stuff. I will have to look into when Photoshop considers a layer a text layer, as I think it deduces it based on the state of the layer and is not hardcoded into the file itself.
